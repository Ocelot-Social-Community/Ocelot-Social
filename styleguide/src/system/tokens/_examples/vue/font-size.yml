#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
