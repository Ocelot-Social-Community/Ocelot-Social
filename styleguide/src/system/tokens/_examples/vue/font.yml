#
# FONT TOKENS
# Use these tokens to set font-weight and font-family.
#

props:
  - name: font-family-heading
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-text
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-code
    value: "inconsolata, monospace"
    category: "font-family"

  - name: font-weight-regular
    value: "normal"
    category: "font-weight"
  - name: font-weight-bold
    value: "600"
    category: "font-weight"
global:
  type: ...
  category: font
