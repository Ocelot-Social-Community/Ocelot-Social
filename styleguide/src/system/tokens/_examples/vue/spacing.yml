#
# SPACING TOKENS
# Use these tokens to set padding, margin and position coordinates.
#

aliases:
  xxx-large:
    value: &xxx-large 120
  xx-large:
    value: &xx-large 60
  x-large:
    value: &x-large 40
  large:
    value: &large 28
  base:
    value: &base 20
  small:
    value: &small 12
  x-small:
    value: &x-small 6
  xx-small:
    value: &xx-small 4
  xxx-small:
    value: &xxx-small 2

props:
  - name: xxx-large
    value: *xxx-large
    category: space-size
  - name: xx-large
    value: *xx-large
    category: space-size
  - name: x-large
    value: *x-large
    category: space-size
  - name: large
    value: *large
    category: space-size
  - name: base
    value: *base
    category: space-size
  - name: small
    value: *small
    category: space-size
  - name: x-small
    value: *x-small
    category: space-size
  - name: xx-small
    value: *xx-small
    category: space-size
  - name: xxx-small
    value: *xxx-small
    category: space-size

  - name: space-xxx-large
    value: "{!xxx-large}px"
  - name: space-xx-large
    value: "{!xx-large}px"
  - name: space-x-large
    value: "{!x-large}px"
  - name: space-large
    value: "{!large}px"
  - name: space-base
    value: "{!base}px"
  - name: space-small
    value: "{!small}px"
  - name: space-x-small
    value: "{!x-small}px"
  - name: space-xx-small
    value: "{!xx-small}px"
  - name: space-xxx-small
    value: "{!xxx-small}px"

global:
  type: number
  category: space
