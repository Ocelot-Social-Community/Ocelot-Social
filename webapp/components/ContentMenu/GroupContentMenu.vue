<template>
  <dropdown class="group-content-menu" :placement="placement" offset="5">
    <template #default="{ toggleMenu }">
      <slot name="button" :toggleMenu="toggleMenu">
        <base-button
          icon="ellipsis-v"
          size="small"
          circle
          @click.prevent="toggleMenu()"
          data-test="group-menu-button"
        />
      </slot>
    </template>
    <template #popover="{ toggleMenu }">
      <div class="group-menu-popover">
        <ds-menu :routes="routes">
          <template #menuitem="item">
            {{ item.parents }}
            <ds-menu-item
              :route="item.route"
              :parents="item.parents"
              @click.stop.prevent="openItem(item.route, toggleMenu)"
            >
              <base-icon :name="item.route.icon" />
              {{ item.route.label }}
            </ds-menu-item>
          </template>
        </ds-menu>
      </div>
    </template>
  </dropdown>
</template>

<script>
import Dropdown from '~/components/Dropdown'

export default {
  name: 'GroupContentMenu',
  components: {
    Dropdown,
  },
  props: {
    usage: {
      type: String,
      required: true,
      validator: (value) => {
        return value.match(/(groupTeaser|groupProfile)/)
      },
    },
    group: { type: Object, required: true },
    placement: { type: String, default: 'bottom-end' },
  },
  computed: {
    routes() {
      const routes = []

      if (this.usage !== 'groupProfile') {
        routes.push({
          label: this.$t('group.contentMenu.visitGroupPage'),
          icon: 'home',
          path: `/groups/${this.group.id}`,
          params: { id: this.group.id, slug: this.group.slug },
        })
      }
      if (this.group.myRole === 'owner') {
        routes.push({
          label: this.$t('admin.settings.name'),
          path: `/groups/edit/${this.group.id}`,
          icon: 'edit',
        })
      }

      return routes
    },
  },
  methods: {
    openItem(route, toggleMenu) {
      if (route.callback) {
        route.callback()
      } else {
        this.$router.push(route)
      }
      toggleMenu()
    },
  },
}
</script>

<style lang="scss">
.group-menu-popover {
  nav {
    margin-top: -$space-xx-small;
    margin-bottom: -$space-xx-small;
    margin-left: -$space-x-small;
    margin-right: -$space-x-small;
  }
}
</style>
