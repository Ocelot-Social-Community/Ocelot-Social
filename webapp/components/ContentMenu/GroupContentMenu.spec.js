import GroupContentMenu from './GroupContentMenu.vue'
import { render, screen, fireEvent } from '@testing-library/vue'

const localVue = global.localVue

const stubs = {
  'router-link': {
    template: '<span><slot /></span>',
  },
  'v-popover': true,
}

// Mock Math.random, used in Dropdown
Object.assign(Math, {
  random: () => 0,
})

describe('GroupContentMenu', () => {
  let mocks

  beforeEach(() => {
    mocks = {
      $t: jest.fn((s) => s),
    }
  })

  const Wrapper = (propsData) => {
    return render(GroupContentMenu, { propsData, mocks, localVue, stubs })
  }

  it('renders as groupTeaser', () => {
    const wrapper = Wrapper({ usage: 'groupTeaser', group: { id: 'groupid' } })
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('renders as groupProfile, not muted', () => {
    const wrapper = Wrapper({
      usage: 'groupProfile',
      group: { isMutedByMe: false, id: 'groupid' },
    })
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('renders as groupProfile, muted', () => {
    const wrapper = Wrapper({
      usage: 'groupProfile',
      group: { isMutedByMe: true, id: 'groupid' },
    })
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('renders as groupProfile when I am the owner', () => {
    const wrapper = Wrapper({
      usage: 'groupProfile',
      group: { myRole: 'owner', id: 'groupid' },
    })
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  describe('mute button', () => {
    it('emits mute', async () => {
      const wrapper = Wrapper({
        usage: 'groupProfile',
        group: { isMutedByMe: false, id: 'groupid' },
      })
      const muteButton = screen.getByText('group.contentMenu.muteGroup')
      await fireEvent.click(muteButton)
      expect(wrapper.emitted().mute).toBeTruthy()
    })
  })

  describe('unmute button', () => {
    it('emits unmute', async () => {
      const wrapper = Wrapper({
        usage: 'groupProfile',
        group: { isMutedByMe: true, id: 'groupid' },
      })
      const muteButton = screen.getByText('group.contentMenu.unmuteGroup')
      await fireEvent.click(muteButton)
      expect(wrapper.emitted().unmute).toBeTruthy()
    })
  })
})
