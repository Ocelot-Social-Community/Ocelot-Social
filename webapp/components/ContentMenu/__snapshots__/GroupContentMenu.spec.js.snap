// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupContentMenu renders as groupProfile when I am the owner 1`] = `
<body>
  <div>
    <v-popover-stub
      autohide="true"
      class="group-content-menu"
      container="body"
      delay="0"
      handleresize="true"
      offset="5"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <button
        class="base-button --icon-only --circle --small"
        data-test="group-menu-button"
        type="button"
      >
        <span
          class="base-icon"
        >
          <!---->
        </span>
         
        <!---->
         
      </button>
       
      <div>
        <div
          class="group-menu-popover"
        >
          <nav
            class="ds-menu"
          >
            <ul
              class="ds-menu-list"
            >
              
          
          
              <li
                class="ds-menu-item ds-menu-item-level-0"
              >
                <a
                  class="ds-menu-item-link"
                  exact="true"
                  href="/"
                >
                  <span
                    class="base-icon"
                  >
                    <!---->
                  </span>
                  
            group.contentMenu.muteGroup
          
                </a>
                <!---->
              </li>
              
          
          
              <li
                class="ds-menu-item ds-menu-item-level-0"
              >
                <a
                  class="ds-menu-item-link"
                  href="/groups/edit/groupid"
                >
                  <span
                    class="base-icon"
                  >
                    <!---->
                  </span>
                  
            admin.settings.name
          
                </a>
                <!---->
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </v-popover-stub>
  </div>
</body>
`;

exports[`GroupContentMenu renders as groupProfile, muted 1`] = `
<body>
  <div>
    <v-popover-stub
      autohide="true"
      class="group-content-menu"
      container="body"
      delay="0"
      handleresize="true"
      offset="5"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <button
        class="base-button --icon-only --circle --small"
        data-test="group-menu-button"
        type="button"
      >
        <span
          class="base-icon"
        >
          <!---->
        </span>
         
        <!---->
         
      </button>
       
      <div>
        <div
          class="group-menu-popover"
        >
          <nav
            class="ds-menu"
          >
            <ul
              class="ds-menu-list"
            >
              
          
          
              <li
                class="ds-menu-item ds-menu-item-level-0"
              >
                <a
                  class="ds-menu-item-link"
                  exact="true"
                  href="/"
                >
                  <span
                    class="base-icon"
                  >
                    <!---->
                  </span>
                  
            group.contentMenu.unmuteGroup
          
                </a>
                <!---->
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </v-popover-stub>
  </div>
</body>
`;

exports[`GroupContentMenu renders as groupProfile, not muted 1`] = `
<body>
  <div>
    <v-popover-stub
      autohide="true"
      class="group-content-menu"
      container="body"
      delay="0"
      handleresize="true"
      offset="5"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <button
        class="base-button --icon-only --circle --small"
        data-test="group-menu-button"
        type="button"
      >
        <span
          class="base-icon"
        >
          <!---->
        </span>
         
        <!---->
         
      </button>
       
      <div>
        <div
          class="group-menu-popover"
        >
          <nav
            class="ds-menu"
          >
            <ul
              class="ds-menu-list"
            >
              
          
          
              <li
                class="ds-menu-item ds-menu-item-level-0"
              >
                <a
                  class="ds-menu-item-link"
                  exact="true"
                  href="/"
                >
                  <span
                    class="base-icon"
                  >
                    <!---->
                  </span>
                  
            group.contentMenu.muteGroup
          
                </a>
                <!---->
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </v-popover-stub>
  </div>
</body>
`;

exports[`GroupContentMenu renders as groupTeaser 1`] = `
<body>
  <div>
    <v-popover-stub
      autohide="true"
      class="group-content-menu"
      container="body"
      delay="0"
      handleresize="true"
      offset="5"
      openclass="open"
      opengroup="0"
      placement="bottom-end"
      popoverarrowclass="tooltip-arrow popover-arrow"
      popoverbaseclass="tooltip popover"
      popoverclass="vue-popover-theme"
      popoverinnerclass="tooltip-inner popover-inner"
      popoverwrapperclass="wrapper"
      popperoptions="[object Object]"
      trigger="manual"
    >
      <button
        class="base-button --icon-only --circle --small"
        data-test="group-menu-button"
        type="button"
      >
        <span
          class="base-icon"
        >
          <!---->
        </span>
         
        <!---->
         
      </button>
       
      <div>
        <div
          class="group-menu-popover"
        >
          <nav
            class="ds-menu"
          >
            <ul
              class="ds-menu-list"
            >
              
          
          
              <li
                class="ds-menu-item ds-menu-item-level-0"
              >
                <a
                  class="ds-menu-item-link"
                  href="/groups/groupid"
                >
                  <span
                    class="base-icon"
                  >
                    <!---->
                  </span>
                  
            group.contentMenu.visitGroupPage
          
                </a>
                <!---->
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </v-popover-stub>
  </div>
</body>
`;
